src/main/java/org/elastos/hive/vendors/onedrive/OneDriveFile.java

```diff
@@ -325,15 +325,21 @@ public String getParentPath() {
 			return future;
 		}
 
-		java.io.File file = CacheHelper.getCacheFile(this.pathName);
-		Length length = new Length(file.length());
+		Length length = new Length(cacheFile.length());
 		future.complete(length);
 		return future;
 	}
 
 	private CompletableFuture<Length> read(Length length, ByteBuffer dest, long position, Callback<Length> callback) {
 		CompletableFuture<Length> future = new CompletableFuture<Length>();
-		if (length.getLength() <= 0) {
+		if (length.getLength() == 0) {
+			Length zero = new Length(0);
+			callback.onSuccess(zero);
+			future.complete(zero);
+			return future;
+		}
+
+		if (length.getLength() < 0) {
```

Comment:
Why do we have this situation ? length.getLength() < 0 ???????

Reply:
Yes, it's a invalid situation. In fact, the file length is not a negative number. i'll delete it.